quint/src/quintVerifier.ts

```diff
@@ -182,7 +184,76 @@ const grpcStubOptions = {
   oneofs: true,
 }
 
-function loadGrpcClient(dist: ApalacheDist): VerifyResult<AsyncCmdExecutor> {
+async function loadProtoDefViaReflection(): Promise<VerifyResult<ProtoPackageDefinition>> {
+  type ServerReflectionResponseSuccess = {
+    file_descriptor_response: {
+      file_descriptor_proto: Buffer[]
+    }
+  }
+  type ServerReflectionResponseFailure = {
+    error_response: {
+      error_code: number
+      error_message: string
+    }
+  }
+  type ServerReflectionResponse = ServerReflectionResponseSuccess | ServerReflectionResponseFailure
+  type ServerReflectionService = {
+    new (url: string, creds: any): ServerReflectionService
+    ServerReflectionInfo: () => any
+  }
+  type ServerReflectionPkg = {
+    grpc: { reflection: { v1alpha: { ServerReflection: ServerReflectionService } } }
+  }
+
+  // Obtain a reflection service client
+  // To load from HTTP:
+  // const protoJson = protobuf.parse(response.data, grpcStubOptions).root.toJSON()
+  // const packageDefinition = proto.fromJSON(protoJson, grpcStubOptions)
+  const protoPath = require.resolve('./reflection.proto');
+  const packageDefinition = proto.loadSync(protoPath, grpcStubOptions);
+  const reflectionProtoDescriptor = grpc.loadPackageDefinition(packageDefinition) as unknown as ServerReflectionPkg
+  const serverReflectionService = reflectionProtoDescriptor.grpc.reflection.v1alpha.ServerReflection
+  const reflectionClient = new serverReflectionService(APALACHE_SERVER_URI, grpc.credentials.createInsecure())
+
+  // Query reflection endpoint
+  const call = reflectionClient.ServerReflectionInfo()
+  call.write({ file_containing_symbol: 'shai.cmdExecutor.CmdExecutor' })
+
+  const protoDefResponse: ServerReflectionResponseSuccess = await new Promise((resolve, reject) => {
+    call.on('data', (reflectionResponse: ServerReflectionResponse) => {
+      if ('file_descriptor_response' in reflectionResponse) {
+        call.cancel()
```

Comment:
On my machine, this `cancel` call triggers the `error` handler below.

Not sure why this uses a streaming API in the first place, but should we just accumulate the responses in a `Buffer[]` and instead resolve the promise in `call`'s `'end'` event?

```
Error: 1 CANCELLED: Cancelled on client
    at callErrorFromStatus (/Users/thomas/src/quint/quint/node_modules/@grpc/grpc-js/build/src/call.js:31:19)
    at Object.onReceiveStatus (/Users/thomas/src/quint/quint/node_modules/@grpc/grpc-js/build/src/client.js:410:73)
    at Object.onReceiveStatus (/Users/thomas/src/quint/quint/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:323:181)
    at /Users/thomas/src/quint/quint/node_modules/@grpc/grpc-js/build/src/resolving-call.js:94:78
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
for call at
    at ServiceClientImpl.makeBidiStreamRequest (/Users/thomas/src/quint/quint/node_modules/@grpc/grpc-js/build/src/client.js:394:34)
    at ServiceClientImpl.<anonymous> (/Users/thomas/src/quint/quint/node_modules/@grpc/grpc-js/build/src/make-client.js:105:19)
    at loadProtoDefViaReflection (/Users/thomas/src/quint/quint/dist/src/quintVerifier.js:138:35)
    at verify (/Users/thomas/src/quint/quint/dist/src/quintVerifier.js:236:19)
    at verifySpec (/Users/thomas/src/quint/quint/dist/src/cliCommands.js:454:39)
    at EitherConstructor.asyncChain (/Users/thomas/src/quint/quint/node_modules/@sweet-monads/either/cjs/index.js:114:16)
    at /Users/thomas/src/quint/quint/dist/src/cli.js:33:39 {
  code: 1,
  details: 'Cancelled on client',
  metadata: Metadata { internalRepr: Map(0) {}, options: {} }
}
```

Reply:
I also have no clue. For now, I pushed a commit without `call.cancel()` and the error is gone. I will look little bit more before I mark this PR as ready-for-review.

Reply:
@thpani apparently it was not a thrown error. rather they are just `console.log` from `error` and `status` callback. I added new commits. these should be printed anymore.

Reply:
btw, it is using streaming API because [the service is defined as streaming](https://github.com/grpc/grpc/blob/498fc99479a01177aa2ad257f1390521495b0788/src/proto/grpc/reflection/v1/reflection.proto#L34-L35).